src/azure-cli/azure/cli/command_modules/acr/task.py

```diff
@@ -401,56 +400,41 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals, too-many-statements
 
     step = task.step
     branch = None
+
+    # If no context is given, use the existing context.
+    # If given context is /dev/null, set it to None.
+    # Otherwise, use the given context.
     if context_path is None:
         context_path = step.context_path
+    elif context_path.lower() == ACR_NULL_CONTEXT:
+        context_path = None
     else:
         branch = _get_branch_name(context_path)
 
+    # When update encoded task, existing encoded_task_content in the step cannot be decoded for update.
+    if context_path is None:
+        if cmd_value is not None or timeout is not None or file is not None:
```

Comment:
I am thinking if customer provide all three of them, we can call get_yaml_template to regenerate the encodedtaskcontent. The logic is similar as create here. If they didn't provide all three, we can just return the error.

Reply:
actually, from what `get_yaml_template` method, we only need cmd_value or file to generate a new yaml_tamplate. Only time we throw an error is when user only wants to update the timeout, which we cannot do because we cannot patch the encoded_task_content, and this is where we are throwing CLIError now.